modules/framework/galasa-parent/dev.galasa.framework.resource.management/src/main/java/dev/galasa/framework/resource/management/internal/ResourceManagement.java

```diff
@@ -161,6 +196,102 @@ public void run(Properties bootstrapProperties, Properties overrideProperties) t
         }
     }
 
+    // Package-level to allow unit testing
+    void loadMonitorBundles(IBundleManager bundleManager, String stream, IStreamsService streamsService) throws FrameworkException {
+        if (stream != null && !stream.isBlank()) {
+            loadRepositoriesFromStream(stream.trim(), streamsService);
+        }
+
+        Set<String> bundlesToLoad = getResourceMonitorBundles();
+
+        // Load the resulting bundles that have the IResourceManagementProvider service
+        for (String bundle : bundlesToLoad) {
+            if (!bundleManager.isBundleActive(bundleContext, bundle)) {
+                logger.info("ResourceManagement - loading bundle: " + bundle);
+
+                bundleManager.loadBundle(repositoryAdmin, bundleContext, bundle);
+
+                logger.info("ResourceManagement - bundle '" + bundle + "' loaded OK");
+            }
+        }
+    }
+
+    private void loadRepositoriesFromStream(String streamName, IStreamsService streamsService) throws FrameworkException {
+        IStream stream = streamsService.getStreamByName(streamName);
+
+        // Add the stream's OBR to the repository admin
+        String commaSeparatedObrs = stream.getObrLocation();
+        if (commaSeparatedObrs == null || commaSeparatedObrs.isBlank()) {
+            throw new FrameworkException("No OBR has been configured into the provided test stream");
+        }
+
+        for (String obr : commaSeparatedObrs.split(",")) {
+            try {
+                repositoryAdmin.addRepository(obr);
+            } catch (Exception e) {
+                throw new FrameworkException("Unable to load repository " + obr, e);
+            }
+        }
+
+        // Add the stream's maven repo to the maven repositories
+        String mavenRepo = stream.getMavenRepositoryUrl();
```

Comment:
Seems this could be put into a stream.isValid() method ?

Reply:
And we could have stream.getMavenRepositoryUrl(); return a URL, with a separate stream.getMavenRepositoryUrlAsString(); then if it wasn't a URL it would blow up before we try to use it, without this code worrying about the MalformedURLException

Reply:
Added isValid() method and changed getMavenRepositoryUrl() return type in https://github.com/galasa-dev/galasa/pull/236/commits/d7bf0fdb58a20eeca9b4d071476e26ee6797180d